### PR TITLE
Privately/purify tabs

### PR DESCRIPTION
**Pull request by walkingice**

### CODE REVIEW

File: app/src/main/java/org/mozilla/rocket/privately/PrivateTabViewProvider.kt [hunk 1]
```diff
@@ -12,6 +13,11 @@ class PrivateTabViewProvider(private val host: Activity) : TabViewProvider {
         return WebViewProvider.create(host, null, WebViewSettingsHook) as TabView
     }
 
+    override fun purify() {
+        create().cleanup()
+        CookieManager.getInstance().removeAllCookies(null)
```
Comment (mxlius) [inline app/src/main/java/org/mozilla/rocket/privately/PrivateTabViewProvider.kt]: Looks like WebkitView.cleanup already includes CookieManager.removeAllCookies. Do we still need to do it here?

Comment (walkingice) [reply app/src/main/java/org/mozilla/rocket/privately/PrivateTabViewProvider.kt]: I was thinking to do that explicitly, but remove that also make sense to me.

File: app/src/main/java/org/mozilla/rocket/privately/PrivateTabViewProvider.kt [hunk 2]
```diff
@@ -12,6 +12,10 @@ class PrivateTabViewProvider(private val host: Activity) : TabViewProvider {
         return WebViewProvider.create(host, null, WebViewSettingsHook) as TabView
     }
 
+    override fun purify() {
+        create().cleanup()
```
Comment (cnevinc) [inline app/src/main/java/org/mozilla/rocket/privately/PrivateTabViewProvider.kt]: Does this create a new Tabview every time a client call purify()?

Comment (walkingice) [reply app/src/main/java/org/mozilla/rocket/privately/PrivateTabViewProvider.kt]: yes, why?

Comment (walkingice) [reply app/src/main/java/org/mozilla/rocket/privately/PrivateTabViewProvider.kt]: Nevin was worrying about performance waste. We will refactor `TabView.cleanup()` soon it should be OK for now.

**Pull request merged**